src/main/java/uk/gov/laa/ccms/caab/service/AddressService.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.laa.ccms.caab.service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.laa.ccms.caab.bean.ClientDetails;
+import uk.gov.laa.ccms.caab.client.OrdinanceSurveyApiClient;
+import uk.gov.laa.ccms.caab.mapper.ClientAddressResultDisplayMapper;
+import uk.gov.laa.ccms.caab.model.ClientAddressResultRowDisplay;
+import uk.gov.laa.ccms.caab.model.ClientAddressResultsDisplay;
+import uk.gov.laa.ccms.caab.model.os.OrdinanceSurveyResponse;
+
+/**
+ * Service class to handle Addresses.
+ */
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class AddressService {
+
+  private final OrdinanceSurveyApiClient ordinanceSurveyApiClient;
+
+  private final ClientAddressResultDisplayMapper clientAddressResultDisplayMapper;
+
+
+  /**
+   * Fetches addresses based on the given postcode by using the Ordinance Survey API.
+   * Maps the API response to ClientAddressResultsDisplay format.
+   *
+   * @param postcode The postcode to search for.
+   * @return ClientAddressResultsDisplay containing a list of addresses associated with the given
+   *         postcode.Returns an empty ClientAddressResultsDisplay if no results are found.
```

Comment:
some weird formatting here

Reply:
A return need to be tabbed like that, to pass checkstyle.